lib/sw_indexer_engine.rb

```diff
@@ -9,17 +9,23 @@ class SwIndexerEngine < BaseIndexer::MainIndexerEngine
   def index(druid, targets = nil)
 
     targets ||= {}
+    update_marc_record_needed = false
 
     ##
     # When a target is not in SKIP_CATKEY_CHECK and its true, check the catkey
-    # If there is a catkey, set false so that the target gets a delete message.
+    # If there is a catkey, set false so that the target gets a delete message and indicate we need a call to the 856 marc generation service.
     # We do this so that the "dor" index doesn't get docs indexed that are
     # already coming from the marc record.
     targets.each do |target_key, target_value|
       next unless target_value == true && !Settings.SKIP_CATKEY_CHECK.include?(target_key)
-      targets[target_key] = false if purl_model(druid).catkey.present?
+      if purl_model(druid).catkey.present?
```

Comment:
Instead of pulling out `update_marc_record_needed`, can you pull out `purl_model(druid)`? It seems like it doesn't depend on the `target` values, right?